File: 12-integer-to-roman/12-integer-to-roman.rb
# @param {Integer} num
# @return {String}
def int_to_roman(num)
    map = {
        'I': 1,
        'IV': 4,
        'V': 5,
        'IX': 9,
        'X': 10,
        'XL': 40,
        'L': 50,
        'XC': 90,
        'C': 100,
        'CD': 400,
        'D': 500,
        'CM': 900,
        'M': 1000
    }
    
    res = ''
    numerals = map.keys.reverse
    numerals.each do |numeral|
        count = num / map[numeral]
        num %= map[numeral]
        res += numeral.to_s * count
    end
    
    res
end

File: medium/12-integer-to-roman/12-integer-to-roman.js
/**
 * @param {number} num
 * @return {string}
 */
const intToRoman = function(num) {
    const numerals = {
        1: 'I', 
        4: 'IV', 
        5: 'V', 
        9: 'IX', 
        10: 'X',
        40: 'XL', 
        50: 'L', 
        90: 'XC', 
        100: 'C', 
        400: 'CD', 
        500: 'D', 
        900: 'CM', 
        1000: 'M'
    }
    let result = '';
    const keys = Object.keys(numerals).reverse();
    console.log(keys)
    keys.forEach(key => {
        while(key<=num) {
            result += numerals[key];
            num -= key
        }
    })
    return result;
};